### PR TITLE
fix: executor/pns containerid prefix fix

### PR DESCRIPTION
**Pull request by L3Nerd**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

In several environments, where docker is used **and** the kubelet and docker are managed by systemd, the cgroup line has a `docker-` prefix. [Here](https://github.com/cri-o/cri-o/issues/842#issuecomment-327428272) is a table of the possible combinations and the resulting cgroup lines.

This PR fixes the behaviour and removes all possible "\*-" prefixes, like "docker-*" in systemd controlled docker/kubelet environments.

Failing example before the fix:
```
time="2020-11-18T12:10:04.158Z" level=info msg="containerID docker-4bb908294439e857f63daec791892a3dddace0cd02bec10fe3f1c5d91589d31f mapped to pid 39"
time="2020-11-18T12:10:04.158Z" level=warning msg="Failed to get main PID: Failed to determine pid for containerID 4bb908294439e857f63daec791892a3dddace0cd02bec10fe3f1c5d91589d31f: container may have exited too quickly"
```

### CODE REVIEW

File: workflow/executor/pns/pns.go [hunk 1]
```diff
@@ -418,18 +418,22 @@ func parseContainerIDFromCgroupLine(line string) string {
...
 			// 9:cpuset:/kubepods-besteffort-pod30556cce_0f92_11eb_b36d_02623cf324c8.slice:cri-containerd:c688c856b21cfb29c1dbf6c14793435e44a1299dfc12add33283239bffed2620
 			if strings.Contains(containerID, "cri-containerd") {
 				strList := strings.Split(containerID, ":")
 				containerID = strList[len(strList)-1]
-				containerID = strings.TrimPrefix(containerID, "cri-containerd-")
 			}
+
+			// remove possible "*-" prefix
+			// e.g. crio-7a92a067289f6197148912be1c15f20f0330c7f3c541473d3b9c4043ca137b42.scope
+			parts := strings.Split(containerID, "-")
```
Comment (alexec) [inline workflow/executor/pns/pns.go:434]: this could panic in `len(parts) == 0`?

Comment (NobbZ) [reply workflow/executor/pns/pns.go:434]: `strings.Split()` always returns a slice of length ≥ 1, as even the empty string, split on any "split char" would result in `[]string{""}`.

Please [refer to the documentation](https://golang.org/pkg/strings/#Split) where it says:

> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

Comment (L3Nerd) [reply workflow/executor/pns/pns.go:434]: An it's already tested [here](https://github.com/argoproj/argo/blob/7a4c487ffaf172542f9ba431cf804822d73d0522/workflow/executor/pns/pns_test.go#L15) and [here](https://github.com/argoproj/argo/blob/7a4c487ffaf172542f9ba431cf804822d73d0522/workflow/executor/pns/pns_test.go#L23).

Comment (alexec) [reply workflow/executor/pns/pns.go:434]: My bad - I actually meant line 435 - but I don't think it can panic on closer inspecting

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
Please sync with master.

**Pull request merged**